### PR TITLE
feat: support flags for `func config git set` command

### PR DESCRIPTION
**Pull request by zroubalik**
# Changes

- :gift: support flags for `func config git set` command

/kind enhancement

Relates: https://github.com/knative/func/issues/1323

```release-notes
Adds support for Git-based deploy options to be configured using CLI flags
```

### CODE REVIEW

File: cmd/config_git_set.go [hunk 1]
```diff
@@ -104,6 +129,8 @@ func (c configGitSetConfig) Prompt(f fn.Function) (configGitSetConfig, error) {
 		return c, err
 	}
 
+	fmt.Printf("%#v\n", c.metadata)
```
Comment (lkingland) [inline cmd/config_git_set.go:132]: This might be a leftover debug statement?

Comment (zroubalik) [reply cmd/config_git_set.go:132]: You are right, good catch! Thanks

File: cmd/config_git_set.go [hunk 2]
```diff
@@ -59,6 +59,25 @@ func NewConfigGitSetCmd(newClient ClientFactory) *cobra.Command {
 		"Specify a custom builder image for use by the builder other than its default. ($FUNC_BUILDER_IMAGE)")
 	cmd.Flags().StringP("image", "i", f.Image, "Full image name in the form [registry]/[namespace]/[name]:[tag]@[digest]. This option takes precedence over --registry. Specifying digest is optional, but if it is given, 'build' and 'push' phases are disabled. (Env: $FUNC_IMAGE)")
 
+	// Git related Flags:
+	cmd.Flags().StringP("git-url", "g", "",
+		"Repo url to push the code to be built (Env: $FUNC_GIT_URL)")
```
Comment (lance) [inline cmd/config_git_set.go:64]: Is code pushed to the repo?
```suggestion
		"Repository url containing the function to build (Env: $FUNC_GIT_URL)")
```

File: cmd/config_git_set.go [hunk 3]
```diff
@@ -59,6 +59,25 @@ func NewConfigGitSetCmd(newClient ClientFactory) *cobra.Command {
...
+	cmd.Flags().StringP("git-branch", "t", "",
+		"Git revision (branch) to be used when deploying via a git repository (Env: $FUNC_GIT_BRANCH)")
+	cmd.Flags().StringP("git-dir", "d", "",
+		"Directory in the repo to find the function (default is the root) (Env: $FUNC_GIT_DIR)")
```
Comment (lance) [inline cmd/config_git_set.go:68]: Nit
```suggestion
		"Directory in the Git repository containing the function (default is the root) (Env: $FUNC_GIT_DIR)")
```

**Review by lance — COMMENTED** [review]
Looks great - just a couple of wordsmithing comments

File: cmd/config_git_set.go (hunk 2 above)
Comment (zroubalik) [reply cmd/config_git_set.go]: Thanks, that's much better. I copy pasted this from `cmd/deploy.go` without proper reading, I fixed it there as well.

**Comment by lance** [comment]
/lgtm
Thanks @zroubalik 👍

**Pull request merged**